test/bot_test.exs

```diff
@@ -34,4 +35,21 @@ defmodule Slack.BotTest do
     assert slack.users    == %{"123" => %{id: "123"}}
     assert slack.ims      == %{"123" => %{id: "123"}}
   end
+
+  defmodule Slack.Rtm.Stub do
+    def start(_token) do
+      {:ok, %{url: "http://www.example.com"}}
+    end
+  end
+
+  defmodule Slack.WebsocketClient do
+    def start_link(_url, _module, _state, _options) do
+      {:ok, self()}
+    end
+  end
+
+  test "can configure the RTM module" do
+    Application.put_env(:slack, :rtm_module, Slack.Rtm.Stub)
+    assert {:ok, _pid} = :"Elixir.Slack.Bot".start_link(Bot, %{}, "token", %{client: Slack.WebsocketClient})
```

Comment:
I'll try removing it, but I think I had a problem at one point with recognizing an embedded module called `Slack` (i.e. From within the module `Slack.MyModule.Slack`, you can't refer to the top level `Slack` module without doing that)... But, for the life of me I can't remember where that was (so itll probably work when I remove it)